internal/volume/volumes_information.go

```diff
@@ -551,6 +551,22 @@ func (v *BackupVolumesInformation) generateVolumeInfoFromPVB() {
 	v.volumeInfos = append(v.volumeInfos, tmpVolumeInfos...)
 }
 
+func (v *BackupVolumesInformation) getVolumeSnapshotClasses(
+	classes *snapshotv1api.VolumeSnapshotClassList,
+) (*snapshotv1api.VolumeSnapshotClassList, error) {
+	if classes != nil {
+		return classes, nil
+	}
```

Comment:
I would do nil check outside this function in `generateVolumeInfoFromDataUpload `.

This function is called `getVolumeSnapshotClasses`, it should be returning the most recent information, rather than exiting.

Also since we are already passing in a pointer, we can just write back to it.
This function shouldn't need any parameter imo unless we rename it to like `getVolumeSnapshotClassesIfNil`.

Reply:
Renaming the function is a good point.

Let me explain why this function looks like this.
The generateVolumeInfoFromDataUpload function is called every time a backup completes. No DataUpload may be included. It's also possible the k8s environment doesn't even have the CSI-related CRDs installed.
Calling to list the VolumeSnapshotClasses in such a condition will raise an unnecessary error.

That's why I moved the VolumeSnapshotClasses listing code from the beginning of the function into the iteration of the operations to make sure only tried to list them when the DataUpload was created.

Reply:
This is what I was thinking.
```diff
-vsClassList, err = v.getVolumeSnapshotClasses(vsClassList)
+if vsClassList == nil {
+	vsClassList, err = v.getVolumeSnapshotClasses()
+	if err != nil {
+		return
+	}
+}
```

This should allow the get func here to be paramless, and still be inside the _iteration of the operations_. But renaming is fine with me too.

Reply:
I see. Thanks.

Reply:
Modified.